Doctutgo - Updated the Go Hello World Tutorial

michael-w-williams opened the pull request:
Let me know what you think. If you like the basic format I will use it as a template for the other hello worlds. Not sure I should show all the variations you can use to call the endpoint.

treeder commented on examples/tutorial/hello/go/README.md:
```diff
@@ -1,66 +1,138 @@
-### First, run the following commands:
+This tutorial assumes you have installed Docker, Fn server, and Fn CLI. See the [Fn Quickstart](https://github.com/fnproject/fn) for installation steps.
+
+## Start Fn Server
```
Maybe just link this to another doc?  If we repeat this everywhere, it will surely get out of sync. Could be part of the assumptions above.

carimura commented:
+1 to let generator create the code.

also you can pass in json by creating a payload.json file and doing `cat payload.json | fn run` which is a nice and elegant demonstration of stdlin and Fn.

shaunsmith commented:
I agree with both Travis and Chad's comments.  I also wonder if we should be focusing on the local development experience with these simple tutorials, or at least calling out the use of --local?  The setting of FN_REGISTRY is covered in the quickstart but may be overlooked.  Helping users get started I've seen it be a stumbling block.

Typo: "It will also demonstrates".

denismakogon commented on examples/tutorial/hello/go/README.md:
````diff
@@ -1,66 +1,138 @@
-# This will create a route to your function as well
-fn deploy --app myapp
-```
+1. Create an empty directory called `hello` and `cd` into it.
````
worth saying that it's a cold function, for "hot" refer to FDK-go

michael-w-williams replied:
Hey Denis. So the purpose of these hello tutorials is to get from nothing to something as quickly as possible. So I don't think a hot v cold discussion is needed for that. But if you think its really really important I can put something in.

denismakogon commented on examples/tutorial/hello/go/README.md:
```diff
@@ -1,66 +1,138 @@
+name: hello
+version: 0.0.1
+runtime: go
+entrypoint: ./func
```
better to add explicitly `type`, `timeout`

michael-w-williams replied:
Is this really needed for  a Hello World example?  As a noob, can you tell more more about this or where I could read more about type and timeout?

denismakogon replied:
Take a look at this doc: https://github.com/fnproject/fn/blob/master/docs/function-file.md
I do think that we should explain what all fields mean and which ones can be omitted in favour of default values.

michael-w-williams commented:
I made most of the requested changes. Also added FN_REGISTRY section as somehow I left that out. Changed deployment example to local. 

So as I see it, these hello tutorials are focused on getting from nothing to a running function as fast as possible. This is a pattern we are using now on other cloud services. For that reason, I didn't add content for fn test or hot v cold functions. If you feel strongly that one of those topics needs to be in here, let me know and we can discuss it further. Thanks.

hibooboo2 commented:
# Oracle CLA Bot
	Thank you for your submission, we really appreciate it. Like many open source projects,
	we ask that you sign our [Contributor License Agreement](http://www.oracle.com/technetwork/community/oca-486395.html) before we can accept your contribution.
	
- @michael-w-williams

hibooboo2 commented:
# Oracle CLA Bot
	Thank you for your submission, we really appreciate it. Like many open source projects,
	we ask that you sign our [Contributor License Agreement](http://www.oracle.com/technetwork/community/oca-486395.html) before we can accept your contribution.
	
- @michael-w-williams

hibooboo2 commented:
# CLA BotThank you for your submission!
It appears that the following authors have not signed our Contributor License Agreement:


* [](https://api.github.com/users/michael-w-williams)


Please do so now by following [this link](http://www.oracle.com/technetwork/community/oca-486395.html). We are working on modernizing the CLA process into a digital signature but it isn’t quite ready yet.
Once complete, let us know in our [community Slack ](slack.fnproject.io) and we’ll send you an Fn T-shirt.
Thank you for being a part of the Fn Community!

hibooboo2 commented:
# CLA Bot
Thank you for your submission!
It appears that the following authors have not signed our Contributor License Agreement:


* [](https://api.github.com/users/michael-w-williams)


Please do so now by following [this link](http://www.oracle.com/technetwork/community/oca-486395.html). We are working on modernizing the CLA process into a digital signature but it isn’t quite ready yet.
Once complete, let us know in our [community Slack ](slack.fnproject.io) and we’ll send you an Fn T-shirt.
Thank you for being a part of the Fn Community!

carimura commented on examples/tutorial/hello/go/README.md:
```diff
@@ -139,7 +151,10 @@ Both commands should return:
 
+## Learn More
+
+* [Documentation](../../../../docs)
```
can we avoid relative links? these will break if the files are moved around

michael-w-williams commented:
Based on Travis' feedback, I dialed the config stuff back and linked to the function-file.md. Added some markdown to make the command match the formatting Shaun is using in the tutorials. Made some other minor fixes.

michael-w-williams commented:
Fixed some conflicts.

treeder commented on examples/tutorial/hello/go/README.md:
```diff
@@ -1,66 +1,185 @@
-# Tutorial 1: Go Function w/ Input (3 minutes)
+# Java Function Hello World
```
This is in the Go directory, but has a bunch of Java stuff and Go stuff mixed together.

michael-w-williams replied:
Let me look.

michael-w-williams replied:
Yep. Fixed it. Also noticed a few other formatting issues. This are all fixed now as well.

michael-w-williams commented:
Made a few more fixes.

shaunsmith commented:
See my feedback for the Java and Node examples which is generically applicable:
- https://github.com/fnproject/fn/pull/698#pullrequestreview-91932335 
- https://github.com/fnproject/fn/pull/697

treeder commented:
@michael-w-williams is this PR still valid?

michael-w-williams commented:
Nope. This is replaced by the language tutorials in the tutorials repo. Closing.
